packages/editor/src/lib/editor/managers/SnapManager.ts

```diff
@@ -159,7 +163,8 @@ function findAdjacentGaps(
 						match.breadthIntersection[1],
 						intersection[0],
 						intersection[1]
-					)!
+					)!,
+					depth + 1
```

Comment:
We use depth here to avoid runaway recursion. I couldn't find the actual cause, so this is the next best thing (afaik depths of >20 are quite rare).